savings-account/src/lib.rs

```diff
@@ -2,8 +2,390 @@
+            self.blockchain()
+                .is_smart_contract(&price_aggregator_address),
+            "Invalid price aggregator address"
+        );
+
+        self.stablecoin_token_id().set(&stablecoin_token_id);
+        self.liquid_staking_token_id().set(&liquid_staking_token_id);
+        self.price_aggregator_address()
+            .set(&price_aggregator_address);
+
+        let pool_params = PoolParams {
+            base_borrow_rate,
+            borrow_rate_under_opt_factor,
+            borrow_rate_over_opt_factor,
+            optimal_utilisation,
+            reserve_factor,
+        };
+        self.pool_params().set(&pool_params);
+
+        Ok(())
+    }
+
+    // endpoints
+
+    #[payable("*")]
+    #[endpoint]
+    fn deposit(
+        &self,
+        #[payment_token] payment_token: TokenIdentifier,
+        #[payment_amount] payment_amount: Self::BigUint,
+    ) -> SCResult<()> {
+        self.require_lend_token_issued()?;
+
+        let stablecoin_token_id = self.stablecoin_token_id().get();
+        require!(
+            payment_token == stablecoin_token_id,
+            "May only deposit stablecoins"
+        );
+
+        let lend_token_id = self.lend_token_id().get();
+        let sft_nonce = self.create_tokens(&lend_token_id, &payment_amount, &())?;
+
+        self.lend_metadata(sft_nonce).set(&LendMetadata {
+            timestamp: self.blockchain().get_block_timestamp(),
+            amount_in_circulation: payment_amount.clone(),
+        });
+
+        let caller = self.blockchain().get_caller();
+        self.send()
+            .direct(&caller, &lend_token_id, sft_nonce, &payment_amount, &[]);
+
+        Ok(())
+    }
+
+    #[payable("*")]
+    #[endpoint]
+    fn borrow(
+        &self,
+        #[payment_token] payment_token: TokenIdentifier,
+        #[payment_nonce] payment_nonce: u64,
+        #[payment_amount] payment_amount: Self::BigUint,
+    ) -> SCResult<()> {
+        self.require_borrow_token_issued()?;
+
+        let liquid_staking_token_id = self.liquid_staking_token_id().get();
+        require!(
+            payment_token == liquid_staking_token_id,
+            "May only use liquid staking position as collateral"
+        );
+
+        let stablecoin_token_id = self.stablecoin_token_id().get();
+        let stablecoin_reserves = self.get_reserves(&stablecoin_token_id);
+        let borrowed_amount = self.borrowed_amount().get();
+
+        let pool_params = self.pool_params().get();
+        let current_utilisation =
+            self.compute_capital_utilisation(&borrowed_amount, &stablecoin_reserves);
+        let borrow_rate = self.compute_borrow_rate(
+            &pool_params.base_borrow_rate,
+            &pool_params.borrow_rate_under_opt_factor,
+            &pool_params.borrow_rate_over_opt_factor,
+            &pool_params.optimal_utilisation,
+            &current_utilisation,
+        );
+
+        let egld_price_in_stablecoin = self.get_egld_price_in_stablecoin()?;
```

Comment:
this could be a little bit more general - no need for egld in the name. As other coins can be integrated later for this savings accounts. so keep like a generalized pair.

Reply:
Valid point. Will try to have a generic "staked token" instead of EGLD specifically.